exercises/saddle-points/.meta/template.j2

```diff
@@ -0,0 +1,46 @@
+"""Tests for the saddle-points exercise
+
+Implementation note:
+The saddle_points function must validate the input matrix and raise a
+ValueError with a meaningful error message if the matrix turns out to be
+irregular.
+"""
+{%- import "generator_macros.j2" as macros with context -%}
+
+{% macro test_case(case) -%}
+    {%- set input = case["input"] -%}
+    {%- set expected = case["expected"] -%}
+    def test_{{ case["description"] | to_snake }}(self):
+        {% if input["matrix"][0] -%}
+        matrix = {{ input["matrix"] }}
+        {% else -%}
+        matrix = []
+        {% endif -%}
+        {% if case is error_case -%}
+        with self.assertRaisesWithMessage(ValueError):
+            {{- case["property"] | to_snake }}(matrix)
+        {% else -%}
+        self.assertEqual(
+        sorted({{ case["property"] | to_snake }}(matrix), key=lambda d: (d["row"],d["column"])),
+        sorted({{ expected }}, key=lambda d: (d["row"],d["column"]))
```

Comment:
Getting cleaner... Since every test case is using this sort call twice, why don't we extract the sorting to a separate function? Something like the following:
```

{% macro test_case(case) -%}
...
self.assertEqual(
    sorted_points({{ case["property"] | to_snake }}(matrix)),
    sorted_points({{ expected}})
)
...
{% endmacro -%}


def sorted_points(point_list):
    return sorted(point_list, key=lambda p: (p["row"], p["column"]))


class {{ exercise | camel_case }}Test(unittest.TestCase):
```

Reply:
That makes sense. Updated!